dot/digest/digest.go

```diff
@@ -229,6 +254,37 @@ func (h *Handler) handleBlockFinalisation(ctx context.Context) {
 	}
 }
 
+// setBABEDigestsOnFinalization is called only when a block is finalised
+// and defines the correct next epoch data and next config data
+func (h *Handler) setBABEDigestsOnFinalization(header *types.Header) error {
+	h.babeConsensusDigestLock.Lock()
+	defer h.babeConsensusDigestLock.Unlock()
+
+	nextEpochData, has := h.nextEpochData[header.Hash()]
+	if has {
+		h.logger.Debugf("setting BABENextEpochData data: %v for hash: %s", nextEpochData, header.Hash())
+		err := h.handleNextEpochData(nextEpochData, header)
+		if err != nil {
+			return err
+		}
+
+		h.nextEpochData = make(map[common.Hash]types.NextEpochData)
```

Comment:
I don't think deleting the map entirely is the right move here, as there might be unfinalized digest info from the next epoch still in the map waiting to be finalized.